test/ao/sparsity/test_kernels.py

```diff
@@ -110,232 +109,204 @@ def test_sparse_qlinear(self):
                     )
 
 
-class TestQuantizedSparseLayers(TestCase):
-    class SparseQuantizedModel(nn.Module):
-        def __init__(self, in_channels, out_channels):
-            super().__init__()
-            self.linear = nn.Linear(in_channels, out_channels)
-
-        def forward(self, x):
-            return self.linear(x)
+def _sparse_layer_test_helper(
+    model_class,
+    sparse_mapping,
+    ref_mapping,
+    qconfig_dict,
+    fqn_to_check,
+    test_class,
+    test_scripting,
+):
+    # SET UP TEST PARAMETERS, INPUTS AND WEIGHTS
+    # ------------------------------------------
+    batch_size = 12
+    input_channels = 4
+    output_channels = 7
+    model = model_class(input_channels, output_channels)
+
+    # For sparse kernels both the activation and weight ZP = 0
+    X_scale = 0.2
+    X_zp = 2
+    W_scale = 1e-2
+    W_zp = 0
+
+    X_fp32 = torch.randn(batch_size, input_channels, dtype=torch.float32)
+    float_bias = torch.randn(output_channels, dtype=torch.float32)
+
+    # generate a weight which we'll insert into the model
+    W_fp32 = torch.randn(output_channels, input_channels, dtype=torch.float32)
+    mask = torch.randint(0, 2, W_fp32.shape)
+    W_fp32 *= mask
+    with override_cpu_allocator_for_qnnpack(qengine_is_qnnpack()):
+        X_q = torch.quantize_per_tensor(
+            X_fp32, scale=X_scale, zero_point=X_zp, dtype=torch.quint8
+        )
+        X_fp32 = X_q.dequantize()
+
+        W_q = torch.quantize_per_tensor(W_fp32, W_scale, W_zp, torch.qint8)
+
+        # PREPARE MODELS FOR QUANTIZATION
+        # -------------------------------
+        model.linear.weight = nn.Parameter(W_q.dequantize())
```

Comment:
so IIUC this step isn't really required, without this the test would just run for some dummy weight in `model.linear`?

Reply:
yeah, i left this in because it seems necessary if we are to eventually refactor this with the kernel tests but you're right, it doesn't do a ton for the layer based tests.